### PR TITLE
Introduce Actions, javascriptExecutor and RemoteWebDriver

### PR DESCRIPTION
**Pull request by AliakseiHlebka**

### CODE REVIEW

File: src/test/java/page/GoogleMailCreateNewEmailPage.java [hunk 1]
```diff
@@ -44,8 +46,10 @@ public GoogleMailCreateNewEmailPage enterEmailAddressee(String email) {
         return this;
     }
 
-    public GoogleMailCreateNewEmailPage enterEmailSubject(String subject) {
-        emailSubjectTextField.sendKeys(subject);
+    public GoogleMailCreateNewEmailPage enterEmailSubject() {
+//        emailSubjectTextField.sendKeys(subject);
+        WebElement element = emailSubjectTextField;
+        javascriptExecutor.executeScript("arguments[0].value='Message from Webdriver';", element);
```
Comment (68889) [inline src/test/java/page/GoogleMailCreateNewEmailPage.java]: Return the method parameter, please

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java [hunk 2]
```diff
@@ -32,26 +58,29 @@ public void googleMailSaveMailAsDraftTest() {
...
         String emailAddressee = "al.hlebka@gmail.com";
         String emailSubject = "Message from Webdriver";
-        String emailBody = "Hello Aliaksei,\nthis message is generated by Webdriver.\nBest regards,\nAliaksei.";
+        JavascriptExecutor javascriptExecutor = (JavascriptExecutor) driver;
 
         GoogleMailHomePage homePage = new GoogleLoginPage(driver).loginToGoogleMail(login, password);
 
-        Assert.assertEquals(driver.getCurrentUrl(), homePage.getHomepageUrl(), "Login failed");
+//        Assert.assertEquals(driver.getCurrentUrl(), homePage.getHomepageUrl(), "Login failed");
+        String currentUrl = (String) javascriptExecutor.executeScript("return document.location.href;");
```
Comment (68889) [inline src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: Move it out of this layer. It should not be placed in the test.

File: src/test/java/test/SimpleJavaScriptExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Test (description = "Smoke")
+    public void testJavaScript() {
+        String login = "tyrmandyr1@gmail.com";
+        String password = "@Tyrmandyr1!";
+
+        GoogleMailHomePage homePage = new GoogleLoginPage(driver).loginToGoogleMail(login, password);
+        String currentUrl = javascriptExecutor.executeScript("return document.location.href;").toString();
+        Assert.assertEquals(currentUrl, homePage.getHomepageUrl(), "Login failed");
+        javascriptExecutor.executeScript("alert('Test Passed!');");
```
Comment (68889) [inline src/test/java/test/SimpleJavaScriptExecutorTest.java]: The same as before

File: src/test/java/test/SimpleJavaScriptExecutorTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        new WebDriverWait(driver, 5).until(ExpectedConditions.alertIsPresent());
+        driver.switchTo().alert().accept();
```
Comment (68889) [inline src/test/java/test/SimpleJavaScriptExecutorTest.java]: The same as before

File: src/test/java/page/GoogleMailCreateNewEmailPage.java (hunk 1 above)
Comment (AliakseiHlebka) [reply src/test/java/page/GoogleMailCreateNewEmailPage.java]: Method parameter (String subject) added, script updated.

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java (hunk 2 above)
Comment (AliakseiHlebka) [reply src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: Method getCurrentPageUrl() created in GoogleMailHomePage class.

File: src/test/java/test/SimpleJavaScriptExecutorTest.java (hunk 3 above)
Comment (AliakseiHlebka) [reply src/test/java/test/SimpleJavaScriptExecutorTest.java]: Method getCurrentPageUrl() reused in Assert..

File: src/test/java/test/SimpleJavaScriptExecutorTest.java (hunk 4 above)
Comment (AliakseiHlebka) [reply src/test/java/test/SimpleJavaScriptExecutorTest.java]: Method generateTestPassAlert() created in GoogleMailHomePage class.

**Pull request merged**